src/modules/SnapperDbus.rb

```diff
@@ -0,0 +1,241 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2015 SUSE LLC. All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "dbus"
+
+module Yast
+
+  class SnapperDbusClass < Module
+
+    include Yast::Logger
+
+    def main
+
+      log.info("connecting to snapperd")
+
+      @system_bus = DBus::SystemBus.instance()
+      @service = @system_bus.service("org.opensuse.Snapper")
+      @dbus_object = @service.object("/org/opensuse/Snapper")
+      @dbus_object.default_iface = "org.opensuse.Snapper"
+      @dbus_object.introspect()
+
+    end
+
+
+    # Escape a String or Hash for snapperd. See snapper dbus documentation for details.
+    def escape(str)
```

Comment:
Then those methods are public just because you need to test them. We had some debate about testing (or not) private methods. I'm not sure if we have reached an agreement, but I try to avoid those situations and test only public (which relies on private ones). Maybe @jreidinger have something to say about that.

Anyway, although it's not a good idea, you could use the '#send' method to avoid private restrictions.